promises-demo-brightscript/components/MainScene.brs

```diff
@@ -0,0 +1,191 @@
+sub init()
+    initOnScreenImage()
+    m.top.observeFieldScoped("simpleResult", "onSimpleResultChange")
+    m.top.observeFieldScoped("chainResult", "onChainResultChange")
+    m.top.observeFieldScoped("parallelResult", "onParallelResultChange")
+
+    'Uncomment the example below to run...
+
+    'Simple single request
+    ' simpleExample("http://ip-api.com/json/")
+
+    'Chain requests
+    ' chainExample()
+
+    'Parallel requests
+    ' parallelExample()
+
+    'Simple single request with error response
+    ' simpleExample("http://invalid--url.com")
+end sub
+
+function networkRequest(url as string, method = "GET" as string, body = {} as object) as object
+    promise = promises_create()
+    task = createObject("roSGNode", "NetworkTask")
+    task.url = url
+    task.method = method
+    task.body = body
+    task.promise = promise
+    task.control = "RUN"
+    return promise
+end function
+
```

Comment:
I'd love to a more "simple" example that just has a single .then or .finally indicating that the flow is done